bin/pycbc_upload_xml_to_gracedb

```diff
@@ -153,22 +198,44 @@ for event in coinc_table:
     xmldoc.childNodes[-1].appendChild(coinc_inspiral_table_curr)
     xmldoc.childNodes[-1].appendChild(coinc_event_map_table_curr)
     xmldoc.childNodes[-1].appendChild(sngl_inspiral_table_curr)
-    ligolw_utils.write_filename(xmldoc, "tmp_coinc_xml_file.xml")
+
+    if args.filename_T050017:
+        ifos = sorted([sngl.ifo for sngl in sngl_inspiral_table_curr])
+        ifos_str = ''.join(ifos)
+        id_str = args.filename_T050017
+        filename_xml = "{}-PyCBC_{}-{:d}-1.xml".format(ifos_str, id_str, time)
+        filename_psd = "{}-PyCBC_{}-{:d}-1_psd.xml.gz".format(ifos_str, id_str, time)
+    else:
+        print("Using default filenames")
+        filename_xml = "tmp_coinc_xml_file.xml"
+        filename_psd = "tmp_psd.xml.gz"
+
+    ligolw_utils.write_filename(xmldoc, filename_xml)
     psd_xmldoc = lal.series.make_psd_xmldoc(psddict)
-    ligolw_utils.write_filename(psd_xmldoc, "tmp_psd.xml.gz", gz=True)
+    ligolw_utils.write_filename(psd_xmldoc, filename_psd, gz=True)
+
+    if args.no_upload:
+        # We have saved the file, and aren't uploading, so reset the tables
+        # and move to the next event
+        xmldoc.childNodes[-1].removeChild(coinc_event_table_curr)
+        xmldoc.childNodes[-1].removeChild(coinc_inspiral_table_curr)
+        xmldoc.childNodes[-1].removeChild(coinc_event_map_table_curr)
+        xmldoc.childNodes[-1].removeChild(sngl_inspiral_table_curr)
+        continue
+
     if args.testing:
-        r = gracedb.createEvent("Test", "pycbc", "tmp_coinc_xml_file.xml",
+        r = gracedb.createEvent("Test", "pycbc", filename_xml,
                                 search="AllSky", offline=True).json()
     else:
-        r = gracedb.createEvent("CBC", "pycbc", "tmp_coinc_xml_file.xml",
+        r = gracedb.createEvent("CBC", "pycbc", filename_xml,
                                 search="AllSky", offline=True).json()
     logging.info("Uploaded event %s.", r["graceid"])
 
     # upload PSD
     gracedb.writeLog(
         r["graceid"], "PyCBC PSD estimate from the time of event",
-        "psd.xml.gz", open("tmp_psd.xml.gz", "rb").read(), "psd").json()
-    logging.info("Uploaded file psd.xml.gz to event %s.", r["graceid"])
+        "psd.xml.gz", open(filename_psd, "rb").read(), "psd").json()
```

Comment:
Should you also replace `"psd.xml.gz"` with `filename_psd`? Genuine question actually, I forget if downstream software expects it to be literally named `psd.xml.gz`!

Reply:
Ah, I hadn't thought of the downstream software relying on things.

I could make it so that the naming convention can only be used with --no-upload?

Reply:
I guess I was trying to say that, with the change you are currently proposing, the file will still be named `psd.xml.gz` on GraceDB, even when using `--format-T050017`. But it sounds like you are thinking of using `--format-T050017` in combination with `--no-upload`, so maybe you are ok with that.

Reply:
looking at the writelog documentation [here](https://ligo-gracedb.readthedocs.io/en/latest/api.html?highlight=writelog#ligo.gracedb.rest.GraceDb.writeLog)

> filecontents (file, optional) – handler pointing to a file to be read and uploaded. If this argument is specified, the contents will be saved as filename on the server.

The bytes object which we get from the read method is the fie contents, and the psd.xml.gz is the filename it is saved as - so it's accidentally (i.e. I forgot to change that bit) the right answer!

Reply:
Looking at the gstlal uploads filenames (i.e. if the requirements are the same, the filenames must be the same), the psds do require the same filename; `psd.xml.gz`, but the coinc xmls do not